msticpy/context/vtlookupv3/vtlookup.py

```diff
@@ -190,7 +197,7 @@ def lookup_iocs(
         src_col: str = "Observable",
         type_col: str = "IoCType",
         src_index_col: str = "SourceIndex",
-        **kwargs,
+        **kwargs: str,
```

Comment:
kwargs shouldn't be a str right?
Maybe you could do a quick sweep for kwargs being mis-typed.

Reply:
I'm also planning to remove (most of) kwargs in a future PR, but in some cases such as this one, I do not know the list of keywords we could have, nor if it's a finite list

Reply:
It's def not treated as a string.  line 256 iterates through the kwargs dict.
Having said that, this is a very old bit of code which I'm not sure anyone uses (vs the TIProvider implementation, so I'd be open to jettisoning...maybe even deprecating the class altogether.

I'm a bit concerned about the idea of removing kwargs everywhere. I think it's OK where they are only used one level deep but sometimes (like data providers), stuff that is unknown to the outer methods gets consumed by something deeper down the stack.

Reply:
Sorry, it seems there is a small misunderstanding there. The [official doc](https://typing.readthedocs.io/en/latest/spec/callables.html#annotating-args-and-kwargs) indicates that type for **kwargs indicates the type of the values from the keyword arguments, not the "real" type of **kwargs which would indeed be a dictionary-like object. Here, **kwargs; int should be understood as dict[str, int].

There are deeper explanations for TypedDict and Unpack that I am not really familiar with yet in that doc that should be leveraged when kwargs cannot be removed, as for dataproviders indeed. I feel it would make the code a bit longer but the extra added value for clarity is worth it. Happy to tone it down if you believe I'm going too far down that rabbit hole.

Reply:
Oh! I didn't know that **kwargs: str meant dict where values are of type str!
Cool. Ok that looks good then.

I'm happy with the replacing kwargs idea where it is pretty straighforward. It def gets trickier when you have a facade class like QueryProvider. 
Often I've used kwargs where I think we might need to extend functionality without messing with the signature. I guess as long as the named args remain optional though, it doesn't cause any backward compat issues (and makes the signatures and docs more understandable.
Happy to discuss on a case-by-case basis.